modules/effects/arrow-effects-rx2-extensions/src/main/kotlin/arrow/effects/rx2/extensions/observablek.kt

```diff
@@ -149,6 +163,83 @@ interface ObservableKEffect : Effect<ForObservableK>, ObservableKAsync {
     fix().runAsync(cb)
 }
 
+interface ObservableKConcurrent : Concurrent<ForObservableK>, ObservableKAsync {
+  override fun <A> async(fa: Proc<A>): ObservableK<A> =
+    ObservableK.async { _, cb -> fa(cb) }
+
+  override fun <A> asyncF(k: ProcF<ForObservableK, A>): ObservableK<A> =
+    ObservableK.asyncF { _, cb -> k(cb) }
+
+  override fun <A> asyncF(fa: ConnectedProcF<ForObservableK, A>): ObservableK<A> =
+    ObservableK.asyncF(fa)
+
+  override fun <A> CoroutineContext.startFiber(kind: ObservableKOf<A>): ObservableK<Fiber<ForObservableK, A>> =
+    asScheduler().let { scheduler ->
+      Observable.create<Fiber<ForObservableK, A>> { emitter ->
+        if (!emitter.isDisposed) {
+          val s: ReplaySubject<A> = ReplaySubject.create()
+          val conn: RxDisposable = kind.value().subscribeOn(scheduler).subscribe(s::onNext, s::onError)
```

Comment:
I'm seeing some `UndeliveredException` here. I tried throwing some synchronization at this, and the `cancel` operation but didn't seem to have any effect. Completely removing `s::onError` also didn't remove the `UndeliveredException` so maybe this is not the culprit.

Another thing I can imagine is an `UndeliveredException` is our `asScheduler` because that seems to throw in `schedule` but haven't gone further down that path.